lib/src/core/room.dart

```diff
@@ -1129,3 +1143,277 @@ extension RoomHardwareManagementMethods on Room {
+    }
+  }
+
+  void handleIncomingRpcResponse(
+    String requestId,
+    String? payload,
+    RpcError? error,
+  ) {
+    final handler = rpcPendingResponses[requestId];
+    if (handler != null) {
+      handler(payload, error);
+      rpcPendingResponses.remove(requestId);
+    } else {
+      logger.warning('Response received for unexpected RPC request $requestId');
+    }
+  }
+
+  Future<void> handleIncomingRpcRequest(
+    String callerIdentity,
+    String requestId,
+    String method,
+    String payload,
+    num responseTimeoutMs,
+    num version,
+  ) async {
+    await publishRpcAck(
+      destinationIdentity: callerIdentity,
+      requestId: requestId,
+    );
+
+    RpcError? responseError;
+    String? responsePayload;
+
+    try {
+      if (version != kRpcVesion) {
+        await publishRpcResponse(
+          destinationIdentity: callerIdentity,
+          requestId: requestId,
+          error: RpcError.builtIn(RpcError.unsupportedVersion).toProto(),
+        );
+        return;
+      }
+
+      var handler = rpcHandlers[method];
+      if (handler == null) {
+        await publishRpcResponse(
+          destinationIdentity: callerIdentity,
+          requestId: requestId,
+          error: RpcError.builtIn(RpcError.unsupportedMethod).toProto(),
+        );
+        return;
+      }
+
+      final response = await handler(RpcInvocationData(
+          requestId: requestId,
+          callerIdentity: callerIdentity,
+          payload: payload,
+          responseTimeoutMs: responseTimeoutMs.toInt()));
+
+      if (response.length > kRpcMaxPayloadBytes) {
+        responseError = RpcError.builtIn(RpcError.responsePayloadTooLarge);
+        logger.warning('RPC Response payload too large for $method');
+      } else {
+        responsePayload = response;
+      }
+    } catch (error) {
+      if (error is RpcError) {
+        responseError = error;
+      } else {
+        logger.warning(
+            'Uncaught error returned by RPC handler for ${method}. Returning RpcError.applicationError instead. $error');
+        responseError = RpcError(
+            code: RpcError.applicationError, message: error.toString());
+      }
+    }
+
+    await publishRpcResponse(
+      destinationIdentity: callerIdentity,
+      requestId: requestId,
+      payload: responsePayload,
+      error: responseError?.toProto(),
+    );
+
+    logger.fine('RPC request ${method} handled');
+  }
+
+  /// Initiate an RPC call to a remote participant.
+  /// @param destinationIdentity - The `identity` of the destination participant
+  /// @param method - The method name to call
+  /// @param payload - The method payload
+  /// @param responseTimeoutMs - Timeout for receiving a response after initial connection
+  /// @returns A promise that resolves with the response payload or rejects with an error.
+  /// @throws Error on failure. Details in `message`.
+  Future<String> performRpc(PerformRpcParams params) async {
```

Comment:
I think the current design has the handler registration on room level, but `performRpc` is still on local_participant, right? 
cc @bcherry

Reply:
oh yeah that's correct